cluster-master.js

```diff
@@ -307,15 +315,23 @@ function forkListener () {
 
     worker.on("disconnect", function () {
       debug("Worker "+id+" disconnect")
-      // give it 1 second to shut down gracefully, or kill
-      disconnectTimer = setTimeout(function () {
-        debug("Worker "+id+", forcefully killing")
-        worker.process.kill("SIGKILL")
-      }, gracefulShutdownDelayMs)
+      startDisconnectTimer(worker, id)
     })
   })
 }
 
+function startDisconnectTimer(worker, id) {
+  clearTimeout(worker.disconnectTimer)
+
+  debug("Worker "+id+" starting disconnect timer")
+
+  // give it 1 second to shut down gracefully, or kill
```

Comment:
nit: this delay is based on `gracefulShutdownDelayMs` we should update the comment to reflect that